xonsh/platform.py

```diff
@@ -0,0 +1,147 @@
+""" Module for platform-specific constants and implementations, as well as
+    compatibility layers to make use of the 'best' implementation available
+    on a platform. """
+
+from functools import lru_cache
+import os
+import platform
+import sys
+
+try:
+    import distro
+except ImportError:
+    distro = None
+
+
+#
+# OS
+#
+
+ON_DARWIN = platform.system() == 'Darwin'
+ON_LINUX = platform.system() == 'Linux'
+ON_WINDOWS = platform.system() == 'Windows'
+
+ON_POSIX = (os.name == 'posix')
+
+
+
+#
+# Python & packages
+#
+
+PYTHON_VERSION_INFO = sys.version_info[:3]
+
+if PYTHON_VERSION_INFO < (3, 5, 0):
+    from pathlib import Path
+
+    def _path(self):
+        return str(self)
+    Path.path = property(_path)
+
+    def scandir(path):
+        """ Compatibility wrapper for os.scandir from Python 3.5+ """
+        return (Path(path) / x for x in os.listdir(path))
+else:
+    from os import scandir
+
+
+ON_ANACONDA = any(s in sys.version for s in {'Anaconda', 'Continuum'})
+
+try:
+    import pygments
+except ImportError:
+    HAS_PYGMENTS, PYGMENTS_VERSION = False, None
+else:
+    HAS_PYGMENTS, PYGMENTS_VERSION = True, pygments.__version__
+
+
+try:
+    import prompt_toolkit
```

Comment:
This also should be a function.  We shouldn't import ptk if we know we are going to use readline.